### PR TITLE
Add pydantic and user input models

### PR DESCRIPTION
**Pull request by alanisaac**
Creating a PR from our work this week, to see CI status and as a reminder for next week.

### CODE REVIEW

File: rules-engine/src/rules_engine/engine.py [hunk 1]
```diff
@@ -1,10 +1,49 @@
...
+class OilPropaneBillingInput(BaseModel):
+    # From Oil-Propane tab
+    period_end_date: date = Field(description="Oil-Propane!B")
+    gallons: float = Field(description="Oil-Propane!C")
+
+
+class NaturalGasBillingInput(BaseModel):
+    # From Natural Gas tab
+    period_end_date: date = Field(description="Natural Gas!B")
+    usage_therms: float = Field(description="Natural Gas!D")
```
Comment (axiomizer) [inline rules-engine/src/rules_engine/engine.py:46]: I think we need either the start date or the number of days in the bill right? To get the daily temperatures.

Do we also want the inclusion override?

**Review by axiomizer — COMMENTED** [review]
Looks good to me besides my one comment

File: rules-engine/src/rules_engine/engine.py [hunk 2]
```diff
@@ -1,47 +1,110 @@
...
+    design_temperature_override: Optional[float] = Field(description="Summary!B7")
+    living_area: float = Field(description="Summary!B10")
+    fuel_type: FuelType = Field(description="Summary!B11")
+    heat_sys_efficiency: float = Field(description="Summary!B12")
+    other_fuel_usage_override: Optional[float] = Field(description="Summary!B16")
+    thermostat_set_point: float = Field(description="Summary!B17")
+    setback_temp: float = Field(description="Summary!B18")
+    setback_hours_per_day: float = Field(description="Summary!B19")
+
+
```
Comment (axiomizer) [inline rules-engine/src/rules_engine/engine.py:26]: Why have the Pydantic models in two places?

**Comment by axiomizer** [comment]
Based on the figma, it looks like we'll need to return UA for every billing period

File: rules-engine/tests/test_rules_engine/test_examples.py [hunk 3]
```diff
@@ -19,69 +26,14 @@
...
-    average_indoor_temperature: float
-    difference_between_ti_and_tbp: float
-    design_temperature: float
-    whole_home_ua: int
-    standard_deviation_of_ua: float
-    avg_heat_load: int
-    max_heat_load: int
-
-
-def validate_usage_date(value):
```
Comment (jkwan2011) [inline rules-engine/tests/test_rules_engine/test_examples.py:53]: Did we lose some validation here, or has this been superseded in the pydantic models?

File: rules-engine/tests/test_rules_engine/test_examples.py [hunk 4]
```diff
@@ -90,30 +42,44 @@ def load_summary(folder: str) -> Summary:
...
+        dates = []
+        temperatures = []
+
+        row: Any
         for row in reader:
-            item = TemperatureDataRecord(
-                date=row["Date"], temperature=row[weather_station]  # type: ignore[arg-type]
-            )
-            result.append(item)
+            dates.append(row["Date"])
```
Comment (jkwan2011) [inline rules-engine/tests/test_rules_engine/test_examples.py:79]: Do we need to pass this to `strptime`?

**Review by jkwan2011 — APPROVED** [review]

**Pull request merged**